tracer/src/Datadog.Trace.Tools.Runner/Checks/ProcessBasicCheck.cs

```diff
@@ -245,11 +274,19 @@ internal static bool CheckRegistry(IRegistryService? registry = null, Version? t
                 bool ok = true;
 
                 // Check that the profiler is properly registered
-                if (tracerVersion == null || tracerVersion < new Version("2.14.0.0"))
+                if (tracerProgramVersion is null)
                 {
                     ok &= CheckClsid(registry, ClsidKey);
                     ok &= CheckClsid(registry, Clsid32Key);
                 }
+                else if (RuntimeInformation.OSArchitecture is Architecture.X64)
+                {
+                    ok &= CheckClsid(registry, ClsidKey);
+                }
+                else if (new Version(tracerProgramVersion) < new Version("2.14.0.0") || RuntimeInformation.OSArchitecture is Architecture.X86)
```

Comment:
`RuntimeInformation.OSArchitecture is Architecture.X86`
This checks if the *current* process is 32 bit. I believe you should check the *target* process.
We can extract that information using _GetProcessBitness in ProcessEnvironmentWindows